### PR TITLE
Adding systemd units and timers.

### PR DESCRIPTION
**Pull request by tonyskapunk**
This is the proposal for issue #129 
- Includes systemd timers and services.
- Reorg of systemd/cron files into `utils` dir.
- Makefile can install cron or systemd files.
- `install` option on Makefile will install systemd files if systemd is detected.

### CODE REVIEW

**Comment by tonyskapunk** [comment]
Example of how timers look once they are enabled:

```bash
$ systemctl list-timers --all | grep -P 'ACT|recap'    
NEXT                         LEFT          LAST                         PASSED    UNIT                         ACTIVATES
Fri 2017-11-03 12:50:00 CDT  1min 12s left Fri 2017-11-03 12:40:02 CDT  8min ago  recap.timer                  recap.service
Sat 2017-11-04 01:00:00 CDT  12h left      n/a                          n/a       recaplog.timer               recaplog.service
n/a                          n/a           Fri 2017-11-03 12:26:19 CDT  22min ago recap-onboot.timer           recap-onboot.service 
```

File: Makefile [hunk 1]
```diff
@@ -23,43 +23,72 @@ MANDIR        ?= $(PREFIX)/share/man
 DOCDIR        ?= $(PREFIX)/share/doc
 SYSCONFDIR    ?= /etc
 CRONDIR       ?= $(SYSCONFDIR)/cron.d
+SYSTEMDDIR    ?= /lib/systemd/system
```
Comment (terackspace) [inline Makefile]: The official path is _/usr/lib_, not _/lib_ - [from the docs](https://www.freedesktop.org/software/systemd/man/systemd.unit.html):
```
/etc/systemd/system/*
/run/systemd/system/*
/usr/lib/systemd/system/*
```
On many modern Linux, /lib is a symlink to /usr/lib now (depending on distro).

File: Makefile [hunk 2]
```diff
@@ -23,43 +23,72 @@ MANDIR        ?= $(PREFIX)/share/man
...
 	@install -dm0755 $(DESTDIR)$(LOGDIR)/recap/snapshots
 
+install-cron: recap.cron
+	@echo "Installing cron job..."
+	@install -Dm0644 src/utils/recap.cron $(DESTDIR)$(CRONDIR)/recap
+
+install-systemd: recap.systemd
+	@echo "Installing systemd timers and services..."
+	@install -Dm0644 src/utils/*.service $(DESTDIR)$(SYSTEMDDIR)/
+	@install -Dm0644 src/utils/*.timer $(DESTDIR)$(SYSTEMDDIR)/
```
Comment (terackspace) [inline Makefile]: the ```install``` will fail to work here if the directory doesn't exist; the -D option means the trailing component has to be named explicitly (by name) as it's being done in install-cron. Here, we'll need to add a line above it:
```
@install -dm0755 $(DESTDIR)$(SYSTEMDDIR)/
@install -Dm0644 src/utils/*.service $(DESTDIR)$(SYSTEMDDIR)/
@install -Dm0644 src/utils/*.timer $(DESTDIR)$(SYSTEMDDIR)/
```

File: src/utils/recap-onboot.service.in [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=recap on boot
+Documentation=man:recap(5) man:recap(8)
+After=network-online.target
+Wants=network-online.target
```
Comment (terackspace) [inline src/utils/recap-onboot.service.in]: Just asking the question - why network-online.target? Is this to ensure the iproute2 utilities will function correctly, or ensure it's in multi-user mode? Perhaps [basic.target](https://www.freedesktop.org/software/systemd/man/systemd.special.html) makes more "generic" sense if we don't absolutely require networking?

File: Makefile (hunk 2 above)
Comment (tonyskapunk) [reply Makefile]: Good catch, I've fixed this in the latest commit

File: Makefile (hunk 1 above)
Comment (tonyskapunk) [reply Makefile]: My bad, I've fixed that in the latest commit.

File: src/utils/recap-onboot.service.in (hunk 3 above)
Comment (tonyskapunk) [reply src/utils/recap-onboot.service.in]: I was thinking more of services like mysql or a webserver running, those services require network so is more likely that recap will obtain logs from those services(when needed) if it runs once the netework is "online" per the target.

File: Makefile (hunk 2 above)
Comment (carlwgeorge) [reply Makefile]: That still won't work.  The `-D` flag requires the full path to the destination file (including the file name), and creates all leading directory components.  Creating the directory first with `install -d` doesn't change that.  You either need to:

Comment (tonyskapunk) [reply Makefile]: Hmm, I may be looking at this wrong, but I see that is copying the files as expected for me:
* installing:
```bash
$ make DESTDIR=test install
Installing scripts...
Installing configuration...
Creating log directories...
Installing man pages...
Installing docs...
for service_file in $( ls src/utils/*.service.in ); do \
    sed -e 's|@BINDIR@|/usr/local/sbin|' ${service_file} \
      > $( echo ${service_file} | sed "s,.in,,"); \
done
Installing systemd timers and services...
Is recommended to enable timers and reload systemd daemon.
```

Then the content inside test looks like this:
```bash
$ find test/
test/
test/var
test/var/log
test/var/log/recap
test/var/log/recap/backups
test/var/log/recap/snapshots
test/etc
test/etc/recap
test/usr
test/usr/local
test/usr/local/share
test/usr/local/share/man
test/usr/local/share/man/man5
test/usr/local/share/man/man5/recap.5
test/usr/local/share/man/man8
test/usr/local/share/man/man8/recap.8
test/usr/local/share/man/man8/recaplog.8
test/usr/local/share/man/man8/recaptool.8
test/usr/local/share/doc
test/usr/local/share/doc/recap
test/usr/local/share/doc/recap/CHANGELOG.md
test/usr/local/share/doc/recap/README.md
test/usr/local/share/doc/recap/COPYING
test/usr/local/sbin
test/usr/local/sbin/recap
test/usr/local/sbin/recaplog
test/usr/local/sbin/recaptool
test/usr/lib
test/usr/lib/systemd
test/usr/lib/systemd/system
test/usr/lib/systemd/system/recap-onboot.timer
test/usr/lib/systemd/system/recaplog.timer
test/usr/lib/systemd/system/recaplog.service
test/usr/lib/systemd/system/recap.timer
test/usr/lib/systemd/system/recap.service
test/usr/lib/systemd/system/recap-onboot.service
```

**Comment by tonyskapunk** [comment]
Last commit fixes the paths on clean and gets rid of the ugly output when installing systemd units.

**Comment by terackspace** [comment]
👍 on 2x new commit changes

**Comment by tonyskapunk** [comment]
I've tested this for quite a while and is working as expected:

1. Logs generated every 10m:
  ```bash
# find  /var/log/recap/ -maxdepth 1 -name '*stat_2*'  | sort
/var/log/recap/netstat_20171116-000001.log
/var/log/recap/netstat_20171116-001000.log
/var/log/recap/netstat_20171116-002001.log
/var/log/recap/netstat_20171116-003001.log
/var/log/recap/netstat_20171116-004000.log
/var/log/recap/netstat_20171116-005001.log
/var/log/recap/netstat_20171116-010001.log
/var/log/recap/netstat_20171116-011001.log
/var/log/recap/netstat_20171116-012001.log
/var/log/recap/netstat_20171116-013001.log
```

1. Timers running at the expected times:
  ```bash
# systemctl list-timers |grep -Pi 'unit|recap'
NEXT                         LEFT          LAST                         PASSED        UNIT                         ACTIVATES
Thu 2017-11-16 10:50:00 CST  2min 51s left Thu 2017-11-16 10:40:00 CST  7min ago      recap.timer                  recap.service
Fri 2017-11-17 01:00:00 CST  14h left      Thu 2017-11-16 01:00:01 CST  9h ago        recaplog.timer               recaplog.service
n/a                          n/a           Tue 2017-11-14 21:23:22 CST  1 day 13h ago recap-onboot.timer           recap-onboot.service
```

1. Logs rotated in the right times:
  ```bash
# grep -i start /var/log/recap/recaplog.log |tail -5
2017-11-12 01:00:01-06:00 [INFO] --- Starting run ---
2017-11-13 01:00:01-06:00 [INFO] --- Starting run ---
2017-11-14 01:00:01-06:00 [INFO] --- Starting run ---
2017-11-15 01:00:01-06:00 [INFO] --- Starting run ---
2017-11-16 01:00:01-06:00 [INFO] --- Starting run ---
```

Will give until the end of the day for any additional comment, then I'll merge this.

**Comment by tonyskapunk** [comment]
Merging this one now.

**Pull request merged**